File: CS/MyXtraGrid/MyGridView/MyGridViewInfo.cs
using DevExpress.XtraGrid.Views.Base;
using DevExpress.XtraGrid.Views.Grid.ViewInfo;
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;

namespace MyXtraGrid
{
    public class MyGridViewInfo : GridViewInfo
    {
        public MyGridViewInfo(DevExpress.XtraGrid.Views.Grid.GridView gridView) : base(gridView) { }

        public int CalcGroupRowTextWidth(RowObjectCustomDrawEventArgs args )
        {
            GridGroupRowInfo rowInfo = args.Info as GridGroupRowInfo;
            if (rowInfo == null)
                return 0;
            return (int)rowInfo.Paint.CalcTextSize(args.Graphics, rowInfo.GroupText, rowInfo.Appearance.Font, new StringFormat(), 0).Width;
        }
    }
}

File: VB/MyXtraGrid/MyGridView/MyGridViewInfo.vb
Imports DevExpress.XtraGrid.Views.Base
Imports DevExpress.XtraGrid.Views.Grid.ViewInfo
Imports System
Imports System.Collections.Generic
Imports System.Drawing
Imports System.Linq
Imports System.Text

Namespace MyXtraGrid
	Public Class MyGridViewInfo
		Inherits GridViewInfo

		Public Sub New(ByVal gridView As DevExpress.XtraGrid.Views.Grid.GridView)
			MyBase.New(gridView)
		End Sub

		Public Function CalcGroupRowTextWidth(ByVal args As RowObjectCustomDrawEventArgs) As Integer
			Dim rowInfo As GridGroupRowInfo = TryCast(args.Info, GridGroupRowInfo)
			If rowInfo Is Nothing Then
				Return 0
			End If
			Return CInt(rowInfo.Paint.CalcTextSize(args.Graphics, rowInfo.GroupText, rowInfo.Appearance.Font, New StringFormat(), 0).Width)
		End Function
	End Class
End Namespace
